src/Core/AuthenticationHelpers/ClientRoleHeaderAuthenticationMiddleware.cs

```diff
@@ -94,7 +94,7 @@ public async Task InvokeAsync(HttpContext httpContext)
             // in an authenticated requests.
             if (isAuthenticatedRequest)
             {
-                clientDefinedRole = httpContext.Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER];
+                clientDefinedRole = httpContext.Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER].ToString();
```

Comment:
are we sure the client role header will exist? What if the TryAdd operation above failed due to reasons other than key being already present?

Reply:
I see that the only other way `TryAdd` fails is if the header dictionary is null:
```csharp
        public static bool TryAdd<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key, TValue value)
        {
            if (dictionary is null)
            {
                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.dictionary);
            }

            if (!dictionary.ContainsKey(key))
            {
                dictionary.Add(key, value);
                return true;
            }

            return false;
        }
```

Then, to answer "can the headerDictionary be null"? It seems the answer is "No" per source code:
 - [DefaultHttpRequest.cs](https://github.com/dotnet/aspnetcore/blob/6ef0fe5cffc59fec5a68275b7178db877d8b9396/src/Http/Http/src/Internal/DefaultHttpRequest.cs#L133-L136)
 ```csharp
    public override IHeaderDictionary Headers
    {
        get { return HttpRequestFeature.Headers; }
    }
```
  - [HttpRequestFeature.cs] referenced by DefaultHttpRequest(https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http/src/Features/HttpRequestFeature.cs)
```csharp
 public HttpRequestFeature()
    {
        Headers = new HeaderDictionary(); // <-- always created.
        Body = Stream.Null;
        Protocol = string.Empty;
        Scheme = string.Empty;
        Method = string.Empty;
        PathBase = string.Empty;
        Path = string.Empty;
        QueryString = string.Empty;
        RawTarget = string.Empty;
    }
```